plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/config/OpenLineageListenerConfig.java

```diff
@@ -0,0 +1,134 @@
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.trino.plugin.openlineage.OpenLineageTransport;
+import io.trino.plugin.openlineage.OpenLineageTrinoFacet;
+import io.trino.spi.resourcegroups.QueryType;
+import jakarta.validation.constraints.NotNull;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+public class OpenLineageListenerConfig
+{
+    private OpenLineageTransport transport = OpenLineageTransport.CONSOLE;
+    private URI trinoURI;
+    private List<OpenLineageTrinoFacet> disabledFacets = ImmutableList.of();
+    private Optional<String> namespace = Optional.empty();
+
+    private Set<QueryType> includeQueryTypes = ImmutableSet.<QueryType>builder()
+            .add(QueryType.ALTER_TABLE_EXECUTE)
+            .add(QueryType.DELETE)
+            .add(QueryType.INSERT)
+            .add(QueryType.MERGE)
+            .add(QueryType.UPDATE)
+            .add(QueryType.DATA_DEFINITION)
+            .build();
+
+    @Config("openlineage-event-listener.transport.type")
+    @ConfigDescription("Type of transport used to emit lineage information.")
+    public OpenLineageListenerConfig setTransport(OpenLineageTransport transport)
+    {
+        this.transport = transport;
+        return this;
+    }
+
+    public OpenLineageTransport getTransport()
+    {
+        return transport;
+    }
+
+    @Config("openlineage-event-listener.trino.uri")
+    @ConfigDescription("URI of trino server. Used for namespace rendering.")
+    public OpenLineageListenerConfig setTrinoURI(URI trinoURI)
+            throws Exception
+    {
+        if (trinoURI.getScheme().isEmpty() || trinoURI.getHost().isEmpty()) {
+            throw new Exception("openlineage-event-listener.trino.uri must be formatted strictly as: ${scheme}://${host}:${port}. Received: " + trinoURI);
+        }
+
+        this.trinoURI = trinoURI;
+        return this;
+    }
+
+    @NotNull
+    public URI getTrinoURI()
+    {
+        return trinoURI;
+    }
+
+    @Config("openlineage-event-listener.trino.include-query-types")
+    @ConfigDescription("Which query types emitted by Trino should generate OpenLineage events. Other query types will be filtered out.")
+    public OpenLineageListenerConfig setIncludeQueryTypes(String includeQueryTypes)
+    {
+        this.includeQueryTypes = new HashSet<>(Arrays.stream(includeQueryTypes.split(","))
+                .map(String::trim)
+                .map(QueryType::valueOf)
+                .toList());
+
+        return this;
+    }
+
+    public Set<QueryType> getIncludeQueryTypes()
+    {
+        return includeQueryTypes;
+    }
+
+    @Config("openlineage-event-listener.disabled-facets")
+    @ConfigDescription("Which facets should be removed from OpenLineage events.")
+    public OpenLineageListenerConfig setDisabledFacets(String disabledFacets)
```

Comment:
I think Airlift has a util methods which allows us to directly inject the enum and we could inject them as `Set<OpenLineageTrinoFacet>` and same for QueryTypes as well. 

Can we add test for `OpenLineageListenerConfig`